reactive-client/src/main/java/org/gluonhq/BasicView.java

```diff
@@ -0,0 +1,189 @@
+        shapeCanvas.getChildren().add( new CircleShape().getRootShape());
+    }
+
+    private void removeSelected(ActionEvent event) {
+
+        Optional.ofNullable(selectedShapeProperty.get())
+                .ifPresent( node -> {
+                    clearSelection();
+                    shapeCanvas.getChildren().remove(node);
+                });
+
+    }
+
+    private void clearAllShapes(ActionEvent event) {
+        shapeCanvas.getChildren().clear();
+        clearSelection();
+    }
+
+    private void clearSelection() {
+        select(null);
+    }
+
+    private void select( Shape shape ) {
+        if ( shape != selectedShapeProperty.get() ) {
+            selectedShapeProperty.set(shape);
+        }
+    }
+
+    private Random random = new Random();
+
+    protected double getRandomX() {
+        return random.nextDouble() * this.getWidth();
+    }
+
+    protected double getRandomY() {
+        return random.nextDouble() * this.getHeight();
+    }
+
+
+
+    private class BaseShape {
+
+        private Shape rootShape;
+        private Point2D delta;
+
+        BaseShape(Supplier<Shape> buildRootShape) {
+
+            rootShape = buildRootShape.get();
+            select(rootShape);
+
+            // select shape on click
+            Disposable disposable = JavaFxObservable.eventsOf(rootShape, MouseEvent.MOUSE_CLICKED)
+                    .subscribe( e -> {
+                        select(rootShape);
+                        e.consume();
+                    });
+
+
+            // dispose "listeners" when shape is removed from the scene
+            JavaFxObservable.changesOf(rootShape.sceneProperty())
+                    .filter( scene -> scene == null )
+                    .subscribe( s -> disposable.dispose());
+
+
+            // calculate delta between shape location and initial mouse position on mouse pressed
+            JavaFxObservable
+                    .eventsOf( rootShape, MouseEvent.MOUSE_PRESSED )
+                    .map( e -> new Point2D( e.getSceneX(), e.getSceneY()))
+                    .subscribe( p -> {
+                        select(rootShape);
+                        Bounds bounds = rootShape.localToScene(rootShape.getLayoutBounds());
+                        delta =  p.subtract( new Point2D(bounds.getMinX(), bounds.getMinY()) );
+                    });
+
+            // User current mouse position and delta to recalculate and set new shape location on mouse dragged
+            JavaFxObservable
+                    .eventsOf( rootShape, MouseDragEvent.MOUSE_DRAGGED )
+                    .map( e -> rootShape.sceneToLocal(e.getSceneX() - delta.getX(),  e.getSceneY() - delta.getY()))
+                    .map( p -> rootShape.localToParent(p))
+                    .subscribe( p -> rootShape.relocate(  p.getX(), p.getY()));
+
+        }
+
+        Shape getRootShape() {
+            return rootShape;
+        }
+
+    }
+
+
+    class RectShape extends BaseShape {
```

Comment:
Retrolambda is having trouble with processing this class. I got the following exception while trying to build for android:

        00:00 ERROR: Failed to run Retrolambda
        java.lang.RuntimeException: Failed to backport class: org/gluonhq/BasicView$RectShape
                at net.orfjackal.retrolambda.Transformers.transform(Transformers.java:129)
                at net.orfjackal.retrolambda.Transformers.transform(Transformers.java:107)
                at net.orfjackal.retrolambda.Transformers.backportClass(Transformers.java:47)
                at net.orfjackal.retrolambda.Retrolambda.run(Retrolambda.java:92)
                at net.orfjackal.retrolambda.Main.main(Main.java:28)
        Caused by: java.lang.ClassFormatError: Illegal field name "org.gluonhq.BasicView$this" in class org/gluonhq/BasicView$RectShape

Extracting the shape classes to standalone outer classes fixes that issue.